sdk/src/settings.rs

```diff
@@ -369,8 +499,20 @@ pub(crate) fn load_settings<P: AsRef<Path>>(settings_path: P) -> Result<()> {
 
 /// Load settings form string representation of the configuration.  Format of configuration must be supplied.
 #[allow(unused)]
+// TODO: when this is removed, remove the additional features (for all supported formats) from the Cargo.toml
+#[deprecated]
 pub fn load_settings_from_str(settings_str: &str, format: &str) -> Result<()> {
-    Settings::from_string(settings_str, format).map(|_| ())
+    Settings::from_string(settings_str, format, None).map(|_| ())
+}
+
+#[allow(unused)]
+pub fn load_settings_from_toml(toml: &str) -> Result<()> {
```

Comment:
The current load_settings function (from file) is private to the crate and was never published, so we can reuse that name for this one and just call it load_settings. If we want to keep the other we should rename it to load_settings_from_file()
if the future is only to support toml, then we don't to specify it in the name.